### PR TITLE
update Node to 6.x LTS

### PR DESCRIPTION
**Pull request by dunn**
The 4.x branch [entered maintenance](https://github.com/nodejs/LTS#lts-schedule1) on 1 April.

### CODE REVIEW

**Comment by krainboltgreene** [comment]
Great call.

**Comment by krainboltgreene** [comment]
Need to make sure everything works with the breaking changes from 4 -> 6. I'm pretty sure it'll be fine.

**Comment by dunn** [comment]
Dunno why Travis' `nvm` is getting confused; that version-string works fine locally.

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-6.7.0
+lts/boron
```
Comment (Gargron) [inline .nvmrc]: Perhaps it expects a number version? Travis version of nvm could be different from yours.. I have not really used nvm before either, so I am not sure.

File: .travis.yml [hunk 2]
```diff
@@ -28,7 +26,7 @@ before_install:
   - sudo apt-get -qq update
   - sudo apt-get -qq install g++-4.8
 install:
-  - nvm install $TRAVIS_NODE_VERSION
+  - nvm install
   - npm install -g npm@3
```
Comment (Gargron) [inline .travis.yml]: If Node 6+ comes with npm >=3, this line can be removed

**Comment by dunn** [comment]
https://github.com/travis-ci/travis-ci/issues/7593

**Comment by dunn** [comment]
The Travis people replied; with `language: ruby` they won't guarantee a modern `nvm`, but I forgot we can just do `6` in the `.nvmrc` instead of specifying the LTS name.

**Review by Gargron — APPROVED** [review]

**Pull request merged**